### PR TITLE
Divider: Improve default behavior when in vertical orientation

### PR DESCRIPTION
**Pull request by mirka**
Follow-up to #38842

## What?

- Conditionally renders the divider as an `inline` element when the `orientation` is `vertical`.
- Improves stories and documentation.

## Why?

This ensures that the divider is visible "as expected" in both orientations when inside a flow layout container (not flex). Initially I was only going to improve the docs to demonstrate proper usage in the flex layout case, but I realized that we can also add a conditional `display` declaration so it just works out of the box.

I'm inclined to classify this as a non-breaking change, because any proper usage of the vertical divider should already be `display: inline` or inside a flex container (in which case the `inline` won't matter).

## Testing Instructions

`npm run storybook:dev` and check out the stories for `Divider`.

- The `vertical` orientation works without a flex container.
- Inside a flex container, it still works as expected as long as it's `stretch` aligned.

### CODE REVIEW

File: packages/components/src/divider/types.ts [hunk 1]
```diff
@@ -24,6 +24,20 @@ export interface OwnProps {
 	marginEnd?: SpaceInput;
 }
 
+interface SeparatorPropsWithAdditionalDescription {
```
Comment (mirka) [inline packages/components/src/divider/types.ts]: Too much? Should I just put it inside `OwnProps`?

Comment (ciampo) [reply packages/components/src/divider/types.ts]: Yeah, I'd personally add it to `OwnProps`. For added clarity, we could at most add a comment where we `omit` the prop from `SeparatorProps` to clarify that we're redefining for better docs? 

Alternatively, we could override it just in Storybook via `argTypes` (although I think I'd prefer the TypeScript solution)

**Review by ciampo — APPROVED** [review]
🎉  🚀 

Much better!

File: packages/components/src/divider/types.ts (hunk 1 above)
Comment (mirka) [reply packages/components/src/divider/types.ts]: Agreed. I guess it's pretty self-evident why it's being overridden.

**Pull request merged**